### PR TITLE
Add Tree Grid search sample

### PR DESCRIPTION
**Pull request by RivaIvanova**

### CODE REVIEW

File: samples/grids/tree-grid/data-searching/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        e.preventDefault();
+        gridRef.current.findPrev(searchText, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+    }
+  }
+
+  return (
+    <div className="container sample">
+      <div className="container vertical">
+        <div style={{ marginBottom: "1rem" }} onKeyDown={searchKeyDown}>
+          <IgrInput name="searchBox" value={searchText} inputOcurred={handleOnSearchChange}>
```
Comment (georgianastasov) [inline samples/grids/tree-grid/data-searching/src/index.tsx:96]: Add checking which icon to show the search icon or clear icon based on whether there is a value entered in the IgrInput.

```tsx
{searchText.length === 0 ? (
                <IgrIconButton
                    key="searchIcon"
                    ref={searchIconRef} 
                    variant="flat"
                    name="search" 
                    collection="material"
                />
            ) : (
                <IgrIconButton
                    key="clearIcon"
                    ref={clearIconRef}
                    variant="flat"
                    name="clear"
                    collection="material"
                    clicked={clearSearch}
                />
            )}
```

File: samples/grids/tree-grid/data-searching/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                variant="flat"
+                name="next"
+                collection="material"
+                clicked={nextSearch}
+              ></IgrIconButton>
+            </div>
+          </IgrInput>
+        </div>
+        <IgrTreeGrid ref={gridRef} data={data} autoGenerate="false" 
+          primaryKey="ID" foreignKey="ParentID" allowFiltering="true" height="480px"
```
Comment (georgianastasov) [inline samples/grids/tree-grid/data-searching/src/index.tsx:137]: Make the height and width of the sample 100%.

width="100%" height="100%"

File: samples/grids/tree-grid/data-searching/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  "dependencies": {
+    "igniteui-dockmanager": "1.14.2",
+    "igniteui-react": "18.6.0",
+    "igniteui-react-core": "18.6.0",
+    "igniteui-react-datasources": "18.6.0",
+    "igniteui-react-grids": "18.6.0",
+    "igniteui-react-inputs": "18.6.0",
+    "igniteui-react-layouts": "18.6.0",
+    "igniteui-webcomponents": "4.7.0",
```
Comment (georgianastasov) [inline samples/grids/tree-grid/data-searching/package.json:24]: Update `igniteui-react` packages version to **18.6.1-alpha.0**, 
`igniteui-dockmanager` package version to **1.14.3** and 
`igniteui-webcomponents` to  **4.8.0**.

**Review by georgianastasov — CHANGES_REQUESTED** [review]

File: samples/grids/tree-grid/data-searching/package.json (hunk 3 above)
Comment (RivaIvanova) [reply samples/grids/tree-grid/data-searching/package.json]: @georgianastasov I updated the `igniteui-react` and `igniteui-react-grids` packages and removed the other ones as they were not needed for this sample.

**Review by georgianastasov — APPROVED** [review]

**Pull request merged**